test/nodes/test_ranker.py

```diff
@@ -1,12 +1,70 @@
 import pytest
 import math
+from unittest.mock import patch
 
 from haystack.errors import HaystackError
 from haystack.schema import Document
 from haystack.nodes.ranker.base import BaseRanker
 from haystack.nodes.ranker.sentence_transformers import SentenceTransformersRanker
 
 
+@pytest.mark.unit
+def test_ranker_preprocess_batch_queries_and_docs_raises():
+    query_1 = "query 1"
+    query_2 = "query 2"
+    docs = [Document(content="dummy doc 1")]
+    with patch("haystack.nodes.ranker.sentence_transformers.SentenceTransformersRanker.__init__") as mock_ranker_init:
+        mock_ranker_init.return_value = None
+        ranker = SentenceTransformersRanker(model_name_or_path="fake_model")
+        with pytest.raises(HaystackError):
```

Comment:
I would like to see a more specific error match:
```suggestion
        with pytest.raises(HaystackError, match="a part of the exception message"):
```